midp/content.js

```diff
@@ -4,10 +4,12 @@
 'use strict';
 
 var Content = (function() {
-  var chRegisteredClassName = null;
-  var chRegisteredStorageID = -1;
-  var chRegisteredID = null;
-  var chRegisteredRegistrationMethod = -1;
+  // Allow configuring these values for MIDlets that don't register
+  // the content handler in time.
```

Comment:
> What happens if a new version of a MIDlet changes one of these values? Will its configuration need to be updated and a new FirefoxOS app version pushed? If so, won't that version not work with the older MIDlet?

Yes, we'll need to update the configuration and the configuration won't work with the older MIDlet. But this is a problem that we already have with many of the configuration options.
Since in this case we have a way to detect the mismatch, in a followup we could ask the user to update the app when needed.

> It seems like what we need is for the FG MIDlet never to launch until the BG MIDlet has finished launching. 

Unfortunately, I think we don't really have a way to tell when the BG MIDlet has finished starting.

> Somewhere you mentioned that this caused a startup perf regression. Do you have info on the method you used and how much it hurt startup time?

We've landed a few improvements over the last months that made the FG MIDlet start earlier, I run the startup benchmark to measure them at the time.